x-pack/plugins/security_solution/public/resolver/store/middleware/resolver_tree_fetcher.ts

```diff
@@ -91,6 +92,25 @@ export function ResolverTreeFetcher(
             parameters: databaseParameters,
           },
         });
+
+        /*
+         * Necessary to handle refresh states where another node besides the origin was selected
+         * If the user has selected another node, but is back to viewing the node list
+         * Refresh will reset to the originID.
+         * This is okay for now, but can be updated if we decide to track selectedNode in panelParameters.
+         */
+        if (currentPanelParameters && currentPanelParameters.panelView !== 'nodes') {
+          const { nodeID } = currentPanelParameters.panelParameters;
+          const urlHasDefinedNode = result.find((node) => node.id === nodeID);
+          api.dispatch({
```

Comment:
When resolver initially loads, how does it know to go to the `origin` initially? Could we do this functionality there instead?

Reply:
the origin position in the graph is just 0,0 , so I think doing it this way is the simplest

Reply:
Gotcha, it's also in the reducer:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/resolver/store/reducer.ts#L27

Also seems like this action is deprecated:

![image](https://user-images.githubusercontent.com/56361221/102531117-d3916200-406f-11eb-8472-97fae6c1b2f6.png)

Should we use this one instead: `userSelectedResolverNode`?

Reply:
Yea, I can try that actually. I'll get back to you once I test it out

Reply:
That action, while selecting the node doesn't actually trigger a panning effect